libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -250,11 +250,59 @@ const luaL_Reg AP_Logger_functions[] = {
     {NULL, NULL}
 };
 
+
+static int add_param(lua_State *L) {
+
+    AP_Float* param = (AP_Float *)luaL_checkudata(L, 1, "AP_Float");
+
+    const char * name = luaL_checkstring(L, 2);
+
+    const uint16_t index = static_cast<uint16_t>(luaL_checkinteger(L, 3));
+    const uint16_t index2 = index + 1;
+
+    float def_value = luaL_checknumber(L, 4);
+
+    AP_Param::Info info[] { { AP_PARAM_FLOAT, name, index,    param, {def_value:def_value}, 0},
+                            { AP_PARAM_NONE,  "",   index2, nullptr,  {group_info:nullptr}, 0 }};
+
+
+    AP_Param::Info *ud = (AP_Param::Info *)lua_newuserdata(L, sizeof(info));
+
+    memcpy(ud, info, sizeof(info));
```

Comment:
Getting this error, not really sure how to fix. it would also be nice to be able to copy abitary length arrays one element at a time to allow adding more than one param at a time.
```
../../libraries/AP_Scripting/lua_bindings.cpp:271:34: error: 'void* memcpy(void*, const void*, size_t)' writing to an object of type 'struct AP_Param::Info' with no trivial copy-assignment; use copy-initialization instead [-Werror=class-memaccess]
  271 |     memcpy(ud, info, sizeof(info));
      | 
```

Reply:
It say to use the copy constructor instead of memcpy

Reply:
I don't think I can because of the way lua user data works?